File: UI/travle_core/lib/src/models/dashboard_activity_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'dashboard_activity_item.g.dart';

/// One row of the admin dashboard's recent-activity feed (mirrors the backend
/// `DashboardActivityItem`). [kind] is a stable discriminator ("Booking",
/// "RoleApplication", "Destination") the UI maps to an icon.
@JsonSerializable()
class DashboardActivityItem {
  DashboardActivityItem({
    required this.kind,
    required this.title,
    required this.description,
    required this.timestamp,
  });

  final String kind;
  final String title;
  final String description;
  final DateTime timestamp;

  factory DashboardActivityItem.fromJson(Map<String, dynamic> json) =>
      _$DashboardActivityItemFromJson(json);

  Map<String, dynamic> toJson() => _$DashboardActivityItemToJson(this);
}

File: Backend/Travle.Model/Responses/DashboardActivityItem.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// One row of the admin dashboard's "recent activity" feed — a merged, most-recent-first list of
    /// platform events (new bookings, role applications, submitted destinations). Names only, never raw
    /// IDs (course §K). <see cref="Kind"/> is a stable discriminator the UI maps to an icon.
    /// </summary>
    public class DashboardActivityItem
    {
        /// <summary>Event category: "Booking", "RoleApplication" or "Destination".</summary>
        public string Kind { get; set; } = string.Empty;

        /// <summary>Short headline, e.g. "New booking".</summary>
        public string Title { get; set; } = string.Empty;

        /// <summary>Human-readable detail, e.g. "Mirza Traveler booked 'Mostar Old Town Walking Tour'".</summary>
        public string Description { get; set; } = string.Empty;

        /// <summary>When the event happened (UTC).</summary>
        public DateTime Timestamp { get; set; }
    }
}
